src/validation.cpp

```diff
@@ -2965,10 +2968,13 @@ bool CheckBlock(const CBlock& block, CValidationState& state, const Consensus::P
     if (block.fChecked)
         return true;
 
-    // Check that the header is valid (particularly PoW).  This is mostly
-    // redundant with the call in AcceptBlockHeader.
-    if (!CheckBlockHeader(block, state, consensusParams, fCheckPOW))
-        return false;
+    // We skip POW checks for genesis block
+    if (block.GetHash() != consensusParams.hashGenesisBlock) {
```

Comment:
I would move this test inside `CheckBlockHeader`

Reply:
I need this if block for `CheckBlockSolution` too, though.

Reply:
`CheckBlockSolution` could also return true if  `block.GetHash() == consensusParams.hashGenesisBlock`

Reply:
That's fair. But the check appears in 3 places instead of 2. Not sure why that is better, tbh.